From 87b4c3f9afea8a189d37de0d382ebbc075774b0c Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 17 Oct 2018 20:05:36 +0300
Subject: [PATCH 0732/1051] net: mvpp2: fix ethtool statistic first line print

ethtool -S (statistic) prints incorrect first register name -
instead of the reg-name the priv-flag srting is printed.

This is occurred in the switch() case ETH_SS_STATS followed by
ETH_SS_PRIV_FLAGS without break.

The fix: add the "break"

Change-Id: Iab57b658a780b4898bfac0df23226d4a48b06ec2
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60524
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 41c67d7ba061..a0a05ec77428 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -1373,6 +1373,7 @@ static void mvpp2_ethtool_get_strings(struct net_device *netdev, u32 sset,
 		for (i = 0; i < ARRAY_SIZE(mvpp2_ethtool_regs); i++)
 			memcpy(data + i * ETH_GSTRING_LEN,
 			       &mvpp2_ethtool_regs[i].string, ETH_GSTRING_LEN);
+		break;
 	case ETH_SS_PRIV_FLAGS:
 		memcpy(data, mvpp22_priv_flags_strings,
 		       ARRAY_SIZE(mvpp22_priv_flags_strings) * ETH_GSTRING_LEN);
-- 
2.17.1

